include/circt/Dialect/Calyx/CalyxStructure.td

```diff
@@ -229,6 +237,37 @@ def InstanceOp : CalyxCell<"instance", [
   }];
 }
 
+def PrimitiveOp : CalyxCell<"primitive", [
+    DeclareOpInterfaceMethods<SymbolUserOpInterface>
+  ]> {
+  let summary = "Calyx Primitive Instance";
+  let description = [{
+    Represents an instance of a Calyx primitive.
+
+    ```mlir
+      %c.in, %c.out = calyx.primitive @c of @MyComponent : i64, i16
```

Comment:
At least for now, this is a HW external primitive, and not a component, right? Maybe spell that out in this example as well?